naga/src/front/wgsl/lower/mod.rs

```diff
@@ -2541,12 +2542,76 @@ impl<'source, 'temp> Lowerer<'source, 'temp> {
                         "select" => {
                             let mut args = ctx.prepare_args(arguments, 3, span);
 
-                            let reject = self.expression(args.next()?, ctx)?;
-                            let accept = self.expression(args.next()?, ctx)?;
+                            let reject_orig = args.next()?;
+                            let accept_orig = args.next()?;
+                            let mut values = [
+                                self.expression_for_abstract(reject_orig, ctx)?,
+                                self.expression_for_abstract(accept_orig, ctx)?,
+                            ];
                             let condition = self.expression(args.next()?, ctx)?;
 
                             args.finish()?;
 
+                            fn expr_ty_res<'a>(
+                                ctx: &'a ExpressionContext<'_, '_, '_>,
+                                expr: Handle<ir::Expression>,
+                            ) -> &'a proc::TypeResolution {
+                                &ctx.typifier()[expr]
+                            }
+                            let diag_deets =
```

Comment:
I don't want a code base full of names like this. Please rename to `diagnostic_details`.

Reply:
Oh, dear, I forgot to change this, and had intended it to be a placeholder name written to keep my brain away from bikeshedding before finalizing logic. Thanks for calling me out!

Reply:
Fixed as suggested with bc5421cf.